packages/ui/src/components/Button/SButton.vue

```diff
@@ -0,0 +1,263 @@
+<script lang="ts">
+export default defineComponent({ name: 'SButton' })
+</script>
+
+<script setup lang="ts">
+import { computed, ComputedRef } from 'vue'
+import { SSpinner } from '@/lib'
+import { BUTTON_ICON_POSITION_VALUES, BUTTON_SIZE_VALUES, BUTTON_TYPE_VALUES, SPINNER_SIZE, SPINNER_WIDTH } from './consts'
+import { ButtonType, ButtonSize, ButtonIconPosition } from './types'
+
+function usePropTypeFilter<T>(value: T, validValues: readonly T[], defaultValue: T): ComputedRef<T> {
+  return computed(() => {
+    if (validValues.includes(value)) {
+      return value
+    }
+
+    return defaultValue
+  })
+}
+
+const props = withDefaults(
+  defineProps<{
+    type?: ButtonType,
+    size?: ButtonSize,
+    nativeType?: 'button' | 'reset' | 'submit',
+    icon?: string,
+    iconPosition?: ButtonIconPosition,
+    rounded?: boolean,
+    disabled?: boolean,
+    loading?: boolean,
+    uppercase?: boolean,
+  }>(),
+  {
+    type: 'secondary',
+    size: 'medium',
+    nativeType: 'button',
+    icon: '',
+    iconPosition: 'left',
+    rounded: false,
+    disabled: false,
+    loading: false,
+    uppercase: false,
+  },
+)
+
+const definitelyType = usePropTypeFilter(props.type, BUTTON_TYPE_VALUES, 'primary')
+const definitelySize = usePropTypeFilter(props.size, BUTTON_SIZE_VALUES, 'medium')
+const definitelyIconPosition = usePropTypeFilter(props.iconPosition, BUTTON_ICON_POSITION_VALUES, 'left')
+const isAction = computed(() => definitelyType.value === 'action')
+</script>
+
+<template>
+  <button
+    :type="nativeType"
+    :class="[
+      's-button',
+      `s-button_type_${definitelyType}`,
+      `s-button_size_${definitelySize}`,
+      `s-button_icon-position_${definitelyIconPosition}`,
+      {
+        's-button_disabled': loading || disabled,
+        'rounded-full': isAction && rounded,
+        'sora-tpg-ch3': definitelySize === 'mini' && uppercase,
+      },
+    ]"
+    :disabled="loading || disabled"
+  >
+    <SSpinner
+      v-if="loading"
+      class="flex-grow absolute"
+      data-testid="spinner"
+      :size="SPINNER_SIZE[definitelySize]"
+      :width="SPINNER_WIDTH[definitelySize]"
+    />
+
+    <span
+      class="s-button__icon"
+      :class="{ 'invisible': loading }"
+      data-testid="icon"
+    >
+      <i
+        v-if="icon"
+        :class="icon"
+      />
+      <slot
+        v-else
+        name="icon"
+      />
+    </span>
+    <span
+      :class="{ 'invisible': loading }"
+      data-testid="text"
+    >
+      <slot v-if="!isAction" />
+    </span>
+  </button>
+</template>
+
+<style lang="scss">
+@use '@/theme';
+
+@include theme.typography-preset-default;
```

Comment:
This mixin shouldn't be used within any component. The purpose of this mixin is to use it to define actual typography classes content, and it should be done once per build, on the library user's side. Components should have only bindings to typographic class names.

Reply:
Is it possible use typography preset styles in style tag then?

Reply:
No, it isn't possible :<
Typography tokens are just class names, and their actual definition is out of components' scope.

Reply:
We can only use them like this:

```vue
<template>
  <div class="sora-tpg-h1">
    Heading 1
  </div>
</template>
```